src/source-es6.js

```diff
@@ -2,25 +2,29 @@
 
 var getScrollMaxY;
 
-if (typeof window.scrollMaxY === 'number') {
-  getScrollMaxY = () => window.scrollMaxY;
-} else {
-  let {body, documentElement: html} = document;
+if(typeof window !== 'undefined') {
+  if (typeof window.scrollMaxY === 'number') {
+    getScrollMaxY = () => window.scrollMaxY;
+  } else {
+    let {body, documentElement: html} = document;
 
-  let getInnerHeight;
+    let getInnerHeight;
 
-  if (typeof window.innerHeight === 'number') {
-    getInnerHeight = () => window.innerHeight;
-  } else {
-    getInnerHeight = () => html.clientHeight || body.clientHeight;
-  }
+    if (typeof window.innerHeight === 'number') {
+      getInnerHeight = () => window.innerHeight;
+    } else {
+      getInnerHeight = () => html.clientHeight || body.clientHeight;
+    }
 
-  let getScrollHeight = () => Math.max(
-    body.scrollHeight,
-    body.offsetHeight,
-    html.clientHeight,
-    html.scrollHeight
-  );
+    let getScrollHeight = () => Math.max(
+      body.scrollHeight,
+      body.offsetHeight,
+      html.clientHeight,
+      html.scrollHeight
+    );
 
-  getScrollMaxY = () => getScrollHeight() - getInnerHeight();
+    getScrollMaxY = () => getScrollHeight() - getInnerHeight();
+  }
+} else {
+  getScrollMaxY = () => void 0;
```

Comment:
In strict-mode scripts and ECMAScript modules, there is no need to use `void 0` because `undefined` cannot be overridden there. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode#Converting_mistakes_into_errors